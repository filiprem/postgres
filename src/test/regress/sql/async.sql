--
-- ASYNC
--

--Should work. Send a valid message via a valid channel name
SELECT pg_notify('notify_async1','sample message1');
SELECT pg_notify('notify_async1','sample_message1',true);
SELECT pg_notify('notify_async1','sample_message1',false);
SELECT pg_notify('notify_async1','');
SELECT pg_notify('notify_async1',NULL);

-- Should fail. Send a valid message via an invalid channel name
SELECT pg_notify('','sample message1');
-- ERROR:  channel name cannot be empty
SELECT pg_notify(NULL,'sample message1');
-- ERROR:  channel name cannot be empty
SELECT pg_notify('notify_async_channel_name_too_long______________________________','sample_message1');
-- ERROR:  channel name too long

--Should work. Valid NOTIFY/LISTEN/UNLISTEN commands
NOTIFY notify_async2;
NOTIFY notify_async2 WITH (collapse off);
NOTIFY notify_async2, '';
NOTIFY notify_async2, '' (collapse on);
NOTIFY notify_async2, '' WITH (collapse off);
LISTEN notify_async2;
UNLISTEN notify_async2;
UNLISTEN *;

-- Should fail. Invalid option syntax
NOTIFY notify_async2, 'test' WITH ( wrong opt );
NOTIFY notify_async2, 'test' (collapse = maybe);


-- Should return zero while there are no pending notifications.
-- src/test/isolation/specs/async-notify.spec tests for actual usage.
SELECT pg_notification_queue_usage();
