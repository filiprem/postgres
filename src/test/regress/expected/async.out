--
-- ASYNC
--
--Should work. Send a valid message via a valid channel name
SELECT pg_notify('notify_async1','sample message1');
 pg_notify 
-----------
 
(1 row)

SELECT pg_notify('notify_async1','sample_message1','unique');
 pg_notify 
-----------
 
(1 row)

SELECT pg_notify('notify_async1','sample_message1','all');
 pg_notify 
-----------
 
(1 row)

SELECT pg_notify('notify_async1','');
 pg_notify 
-----------
 
(1 row)

SELECT pg_notify('notify_async1',NULL);
 pg_notify 
-----------
 
(1 row)

-- Should fail. Send a valid message via an invalid channel name
SELECT pg_notify('','sample message1');
ERROR:  channel name cannot be empty
SELECT pg_notify(NULL,'sample message1');
ERROR:  channel name cannot be empty
SELECT pg_notify('notify_async_channel_name_too_long______________________________','sample_message1');
ERROR:  channel name too long
-- Should fail. Invalid 3rd parameter
NOTIFY notify_async2, 'test', 'invalid';
ERROR:  ?
NOTIFY notify_async2, 'test', true;
ERROR:  ?
--Should work. Valid NOTIFY/LISTEN/UNLISTEN commands
NOTIFY notify_async2;
NOTIFY notify_async2, '';
NOTIFY notify_async2, '', 'unique';
NOTIFY notify_async2, '', 'all';
LISTEN notify_async2;
UNLISTEN notify_async2;
UNLISTEN *;
-- Should return zero while there are no pending notifications.
-- src/test/isolation/specs/async-notify.spec tests for actual usage.
SELECT pg_notification_queue_usage();
 pg_notification_queue_usage 
-----------------------------
                           0
(1 row)

